lib/queueservice.js

```diff
@@ -0,0 +1,82 @@
+import xml2js from 'xml2js';
+import Debug from 'debug';
+import request from 'superagent-promise';
+
+let debug = Debug('taskcluster-docker-worker:queueService');
+
+export default class TaskQueue {
+  constructor(config) {
+    this.queues;
+    this.workerType = config.workerType;
+    this.provisionerId = config.provisionerId;
+    this.client = config.queue;
+    this.log = config.log;
+  }
+
+  async getQueues() {
+    if (!this.queues || (new Date(this.queues.expires)).getTime() > Date.now()) {
+      this.queues = await this.client.pollTaskUrls(this.provisionerId, this.workerType);
+    }
+    return this.queues.queues;
+  }
+
+  async getTasksFromQueue(queue, numberOfTasks) {
```

Comment:
Perhaps an even better thing to do would be to move the `claimTask` part from the `task_listener.js` into the `queueservice.js`... Then this file could encapsulate all the code that deals with pulling azure messages, iterating over azure queues and claiming tasks based on messages.

The task listener also deals with other stuff like, waiting for capacity to be available, scheduling polling for tasks and listening for cancels. 

The main reason I suggested exposing integers instead of objects was to reduce the interface between the files. Encapsulation is key and javascript one has to be careful to make sure it gets done. Putting all the polling logic and claim task logic in the same place makes sense. While still have the scheduling for polling of tasks in the task_listener. (my two cents)